src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -1019,6 +1019,34 @@ void loop_inst::set_memory_in_body_network(cldnn::network::ptr body_network,
     }
 }
 
+void loop_inst::set_memory_in_body_network_from_output(cldnn::network::ptr body_network,
```

Comment:
I think if you set body network input layout to dynamic shape in ops/loop.cpp as you put the code in this commit and change set_memory_in_body_network as follows, you don't need to add new method.

``
```suggestion
void loop_inst::set_memory_in_body_network(cldnn::network::ptr body_network,
                const std::shared_ptr<cldnn::primitive_inst>& inst, memory::ptr mem) {
    if (inst->is_input()) {
        // 0. remove interpret code
        body_network->set_input_data(inst->id(), updated_mem);// 1. set_input_data
        inst->update_shape(); // 2. update_shape
    } else if (inst->is_output()) {
        body_network->set_output_memory(inst->id(), mem);
    } else {
        inst->set_output_memory(mem, false);
    }
}
```